server/src/main/kotlin/org/javacs/kt/externalsources/KlsURI.kt

```diff
@@ -29,24 +27,56 @@ fun URI.toKlsURI(): KlsURI? = when (scheme) {
  * Other file extensions for classes (such as .kt and .java) are supported too, in
  * which case the file will directly be used without invoking the decompiler.
  */
-data class KlsURI(val uri: URI) {
+data class KlsURI(val fileUri: URI, val query: Map<QueryParams, Any>) {
+
+    /**
+     * Possible KLS URI query parameters
+     */
+    enum class QueryParams(val parameterName: String) {
+        SOURCE("source");
+
+        override fun toString(): String = parameterName
+
+        companion object {
+            val ValueParsers: Map<QueryParams, (String) -> Any> = mapOf(
```

Comment:
I'd prefer `VALUE_PARSERS`, for consistency (or `valueParsers`, if it feels to 'heavy' for a constant to you).